chef-config/lib/chef-config/config.rb

```diff
@@ -710,6 +711,64 @@ def self.windows_home_path
     config_context :chefdk do
     end
 
+    configurable(:http_proxy)
+    configurable(:http_proxy_user)
+    configurable(:http_proxy_pass)
+    configurable(:https_proxy)
+    configurable(:https_proxy_user)
+    configurable(:https_proxy_pass)
+    configurable(:ftp_proxy)
+    configurable(:ftp_proxy_user)
+    configurable(:ftp_proxy_pass)
+    configurable(:no_proxy)
+
+    # Public method that users should call to export proxies to the appropriate
+    # environment variables.  This method should be called after the config file is
+    # parsed and loaded.
+    # TODO add some post-file-parsing logic that automatically calls this so
+    # users don't have to
+    def self.export_proxies
+      export_proxy("http", http_proxy, http_proxy_user, http_proxy_pass) if http_proxy
+      export_proxy("https", https_proxy, https_proxy_user, https_proxy_pass ) if https_proxy
+      export_proxy("ftp", ftp_proxy, ftp_proxy_user, ftp_proxy_pass ) if ftp_proxy
+      export_no_proxy("no_proxy", no_proxy) if no_proxy
+    end
+
+    # Builds a proxy uri and exports it to the appropriate environment variables. Examples:
+    #   http://username:password@hostname:port
+    #   https://username@hostname:port
+    #   ftp://hostname:port
+    # when
+    #   scheme = "http", "https", or "ftp"
+    #   hostport = hostname:port or scheme://hostname:port
+    #   user = username
+    #   pass = password
+    def self.export_proxy(scheme, path, user, pass)
+      path = "#{scheme}://#{path}" unless path.include?('://')
+      # URI.split returns the following parts:
+      # [scheme, userinfo, host, port, registry, path, opaque, query, fragment]
+      parts = URI.split(URI.encode(path))
+      # URI::Generic.build requires an integer for the port, but URI::split gives
+      # returns a string for the port.
+      parts[3] = parts[3].to_i if parts[3]
+      if user
```

Comment:
```
[chef]$ rando= bundle exec pry
[1] pry(main)> ENV['rando']
=> ""
```

Normally if it isn't specified, `ENV['rando']` will be nil.  But because it looks pretty easy to pass them in as empty strings, I'll convert empty strings I find to nil values.  Thanks!